examples/text-generation/textgen_pipeline/pipeline.py

```diff
@@ -0,0 +1,219 @@
+import copy
+import os
+
+import habana_frameworks.torch.hpu as torch_hpu
+import torch
+from habana_frameworks.torch.distributed.hccl import initialize_distributed_hpu
+from habana_frameworks.torch.hpu import wrap_in_hpu_graph
+from transformers import (
+    AutoConfig,
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    TextGenerationPipeline,
+)
+
+from optimum.habana.checkpoint_utils import (
+    get_ds_injection_policy,
+    get_repo_root,
+    model_is_optimized,
+    model_on_meta,
+    write_checkpoints_json,
+)
+from optimum.habana.transformers.modeling_utils import adapt_transformers_to_gaudi
+from optimum.habana.utils import set_seed
+
+
+class GaudiTextGenerationPipeline(TextGenerationPipeline):
+    """
+    An end-to-end text-generation pipeline that can used to initialize LangChain classes. It supports both single-hpu and multi-hpu inference.
+    """
+
+    def __init__(self, model_name_or_path=None, use_bf16=True, **kwargs):
```

Comment:
```suggestion
    def __init__(self, model_name_or_path=None, bf16=True, **kwargs):
```
Let's call this arg `bf16` to stay consistent with other parts of the codebase